cds-hook-model/src/main/java/io/elimu/a2d2/oauth/OAuthUtils.java

```diff
@@ -0,0 +1,101 @@
+package io.elimu.a2d2.oauth;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.io.StringWriter;
+import java.net.URL;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import org.apache.commons.codec.digest.Md5Crypt;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+
+public class OAuthUtils {
+	
+	private static final Logger LOG = LoggerFactory.getLogger(OAuthUtils.class);
+	private static final Map<String, Map<String, Object>> CACHE = new HashMap<>();
+
+	//TODO invoke this from QueryingServerHelper
+	public static Map<String, Object> authenticate(String body, String tokenUrl, String clientId, String clientSecret) {
+		String keyUnhashed = "url=" + tokenUrl + "&" + body + "&client_id=" + clientId + "&client_secret=" + clientSecret;
```

Comment:
We may need more than `client_secret` here. In cases where there is no client secret, but there is a `username` and `password`, do we want the cache to serve up the same token for different credentials?

Reply:
@hankwallace in those cases, the content will be coming from the body variable, which is built with BodyBuilder using all parameters available for the authentication